### PR TITLE
Fix using Atom with MSVC

### PR DESCRIPTION
**Pull request by KindDragon**

### CODE REVIEW

File: folly/detail/Futex.h [hunk 1]
```diff
@@ -44,7 +44,7 @@ enum class FutexResult {
  */
 template <template <typename> class Atom = std::atomic>
 struct Futex : Atom<uint32_t> {
-  using Atom<uint32_t>::Atom;
+  explicit constexpr Futex(uint32_t init = 0) : Atom<uint32_t>(init) {}
```
Comment (yfeldblum) [inline folly/detail/Futex.h]: Define both ctors listed here, with no default arguments.

http://en.cppreference.com/w/cpp/atomic/atomic/atomic

**Comment by KindDragon** [comment]
Fixed

**Comment by KindDragon** [comment]
MSVC bug https://developercommunity.visualstudio.com/content/problem/60185/cannot-compile-template-class-inheriting-construct.html

**Comment by yfeldblum** [comment]
Can also add a link to this PR into that MSVC bug report.

**Pull request closed without merging**